src/extension/utils.ts

```diff
@@ -56,6 +60,41 @@ const fixNotebookUri = (uri: vscode.Uri) => {
  */
 export const getMalloyConfig = (): vscode.WorkspaceConfiguration => {
   const malloyConfig = vscode.workspace.getConfiguration('malloy');
+
+  // possibly update bigquery config parameters
+  const connectionConfigs = malloyConfig.get(
+    'connections'
+  ) as ConnectionConfig[];
+
+  // if bigquery connection config still uses old "projectName" key,
+  // move over to "projectId", delete the "projectName" key, and save
+  connectionConfigs.forEach(connectionConfig => {
+    if (connectionConfig.backend === ConnectionBackend.BigQuery) {
+      const oldProjectNameValue = (connectionConfig as any)['projectName'];
+
+      if (oldProjectNameValue) {
+        connectionConfig.projectId = oldProjectNameValue;
+        delete (connectionConfig as any)['projectName'];
+
+        const hasWorkspaceConfig =
+          malloyConfig.inspect('connections')?.workspaceValue !== undefined;
+
+        malloyConfig.update(
+          'connections',
+          connectionConfigs,
+          vscode.ConfigurationTarget.Global
+        );
+        if (hasWorkspaceConfig) {
+          malloyConfig.update(
+            'connections',
+            connectionConfigs,
+            vscode.ConfigurationTarget.Workspace
+          );
+        }
```

Comment:
I realize this is just mimicking what we currently do, but it's weird to me that we always write to the global config even if there's a workspace config? It seems like it should be one or the other.

Reply:
Agreed, i'm not sure why it's like this originally. My take is connection info is global setting and we shouldn't care about workspace but perhaps there's some nuance i'm missing. @christopherswenson any thoughts?

Reply:
I can't remember exactly why it's like that right now, but I think the idea was to enable people to just have separate connection settings per workspace if they wanted to, and to update the workspace settings if they had them. I assume it was a mistake to also set global config if we're setting workspace config.

Maybe the idea was that it might be confusing if I did have workspace settings and updated them, but for some reason expected them to update everywhere else, too. But that seems unlikely.

Reply:
Added issue and will try to address once I get Carlin/Lloyd to align on answer: https://github.com/malloydata/malloy-vscode-extension/issues/374